portal-frontend/src/app/services/application-submission/application-submisison-document-generation/application-submission-document-generation.service.ts

```diff
@@ -0,0 +1,45 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import moment from 'moment';
+import { firstValueFrom } from 'rxjs';
+import { environment } from '../../../../environments/environment';
+import { OverlaySpinnerService } from '../../../shared/overlay-spinner/overlay-spinner.service';
+import { getPdfFile } from '../../../shared/utils/file';
+import { ToastService } from '../../toast/toast.service';
+
+interface Report {
+  fileName: string;
+  data: Blob;
+}
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ApplicationSubmissionDocumentGenerationService {
```

Comment:
Do you think its better to have a service for each type or maybe a general DocumentGenerationService that has generateSubmissionPdf, generateReviewPdf, etc?

Reply:
I prefer service per document type since that way it will be easier to maintain them. For example, we will have +-8 submission types and potentially each one of them with a custom payload.